domenetjenester/mottak/src/main/java/no/nav/ung/sak/mottak/dokumentmottak/inntektrapportering/DokumentMottakerInntektrapportering.java

```diff
@@ -0,0 +1,111 @@
+        for (MottattDokument dokument : mottattDokument) {
+            var søknad = søknadParser.parseSøknad(dokument);
+            dokument.setBehandlingId(behandlingId);
+            dokument.setInnsendingstidspunkt(søknad.getMottattDato().toLocalDateTime());
+            if (søknad.getKildesystem().isPresent()) {
+                dokument.setKildesystem(søknad.getKildesystem().get().getKode());
+            }
+            // Setter dokument til BEHANDLER for at AsyncAbakusLagreOpptjeningTask skal plukke den opp (brukes som idempotens-sjekk for kall mot abakus)
+            mottatteDokumentRepository.lagre(dokument, DokumentStatus.BEHANDLER);
+            opprettHistorikkinnslagForVedlegg(behandling.getFagsakId(), dokument.getJournalpostId());
+            lagreOppgittOpptjeningFraSøknad(søknad, behandling, dokument);
+        }
+    }
+
+    /**
+     * Lagrer oppgitt opptjening til abakus fra mottatt dokument.
+     */
+    private void lagreOppgittOpptjeningFraSøknad(Søknad søknad, Behandling behandling, MottattDokument dokument) {
+        try {
+            var inntekter = ((Ungdomsytelse) søknad.getYtelse()).getInntekter();
+            var request = OppgittOpptjeningMapper.mapRequest(BehandlingReferanse.fra(behandling), dokument, inntekter);
+            if (request.map(OppgittOpptjeningMottattRequest::getOppgittOpptjening).isEmpty()) {
+                // Ingenting mer som skal lagres - dokument settes som ferdig
+                mottatteDokumentRepository.oppdaterStatus(List.of(dokument), DokumentStatus.GYLDIG);
+                return;
+            }
+            var enkeltTask = ProsessTaskData.forProsessTask(AsyncAbakusLagreOpptjeningTask.class);
+            var payload = JsonObjectMapper.getMapper().writeValueAsString(request.get());
+            enkeltTask.setPayload(payload);
+
+            enkeltTask.setProperty(AsyncAbakusLagreOpptjeningTask.JOURNALPOST_ID, dokument.getJournalpostId().getVerdi());
+            enkeltTask.setProperty(AsyncAbakusLagreOpptjeningTask.BREVKODER, dokument.getType().getKode());
+
+            enkeltTask.setBehandling(behandling.getFagsakId(), behandling.getId(), behandling.getAktørId().getAktørId());
+            enkeltTask.setSaksnummer(behandling.getFagsak().getSaksnummer().getVerdi());
+            enkeltTask.setCallIdFraEksisterende();
+
+            prosessTaskTjeneste.lagre(enkeltTask);
+        } catch (IOException e) {
+            throw AbakusInntektArbeidYtelseTjenesteFeil.FEIL.feilVedKallTilAbakus("Opprettelse av task for lagring av oppgitt opptjening i abakus feiler.", e).toException();
+        }
+    }
+
+
+    private void opprettHistorikkinnslagForVedlegg(Long fagsakId, JournalpostId journalpostId) {
+        historikkinnslagTjeneste.opprettHistorikkinnslagForVedlegg(fagsakId, journalpostId);
+    }
+
+    @Override
+    public BehandlingÅrsakType getBehandlingÅrsakType(Brevkode brevkode) {
+        return BehandlingÅrsakType.RE_ENDRING_FRA_BRUKER;
```

Comment:
Skal vi istedet bruke RE_RAPPORTERING_INNTEKT her for å være enda mer spesifikk?